src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -2701,7 +2701,7 @@ protected override LocalState UnreachableState()
         protected override LocalState ReachableBottomState()
         {
             // Create a reachable state in which all variables are known to be non-null.
-            return LocalState.ReachableStateWithNotNulls(_variables);
+            return LocalState.StateWithNotNulls(_variables, reachable: true);
```

Comment:
>_variables [](http://example.com/codeflow?start=48&length=10)

It looks like there is a possible flaw in this implementation. 
The doc comment for `ReachableBottomState` states: "The result 
should be a state, for each variable, that is the strongest result 
possible (i.e. definitely assigned for the data flow passes, or 
**not null for the nullable analysis**)."
However, for nullable analysis, the notion of a variable is recursively 
expanded to properties and to fields for classes (fields for structures 
are always recursively tracked), etc. In a situation when some of those 
members haven't been touched yet, we don't have a slot for them, 
thus they won't be marked as `not null`.  We were able to observe that in 
the repro scenario. It is very likely that scenarios that rely on `ReachableBottomState` 
exhibit problems similar to the one we are trying to fix for local functions. 
We don't have to confirm/address that in context of this PR, but, I think, 
at least, we should open an issue to follow up.

Reply:
Filed https://github.com/dotnet/roslyn/issues/71602.